operator/controllers/internal/lokistack/ruler_config_discovery.go

```diff
@@ -0,0 +1,42 @@
+package lokistack
+
+import (
+	"context"
+	"time"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	lokistackv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AnnotateForRulerConfig adds/updates the `loki.grafana.com/rulerConfigDiscoveredAt` annotation
+// to the named Lokistack in the same namespace of the RulerConfig. If no LokiStack is found, then
+// skip reconciliation.
+func AnnotateForRulerConfig(ctx context.Context, k k8s.Client, name, namespace string) error {
```

Comment:
Does it make sense to refactor the `Annotate` functions, so that more code is shared?

Reply:
IMHO the two annotate functions differ in regard of which LokiStack resource they annotate:
- We annotate for discovered rules **all** LokiStack instances on the cluster
- We annotate for a discovered ruler config **only** the LokiStack instance with the same name and on the same namespace as the ruler config.

Reply:
I don't see the benefits to melt helper functions together as it creates more common helper functions for the helper functions :dancers: